services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/HankeMapperTest.kt

```diff
@@ -0,0 +1,92 @@
+package fi.hel.haitaton.hanke
+
+import assertk.all
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import assertk.assertions.isNull
+import assertk.assertions.prop
+import fi.hel.haitaton.hanke.ContactType.MUU
+import fi.hel.haitaton.hanke.ContactType.OMISTAJA
+import fi.hel.haitaton.hanke.ContactType.RAKENNUTTAJA
+import fi.hel.haitaton.hanke.ContactType.TOTEUTTAJA
+import fi.hel.haitaton.hanke.domain.Hanke
+import fi.hel.haitaton.hanke.domain.Hankealue
+import fi.hel.haitaton.hanke.factory.AlluDataFactory.Companion.teppoEmail
+import fi.hel.haitaton.hanke.factory.HankeFactory
+import fi.hel.haitaton.hanke.factory.HankeYhteystietoFactory.create
+import fi.hel.haitaton.hanke.factory.TEPPO_TESTI
+import fi.hel.haitaton.hanke.geometria.Geometriat
+import org.junit.jupiter.api.Test
+
+private const val MOCK_ID = 1
+
+class HankeMapperTest {
+
+    val geometry: Geometriat =
+        "/fi/hel/haitaton/hanke/geometria/hankeGeometriat.json".asJsonResource()
+
+    @Test
+    fun `when entity contains all fields should map domain object correspondingly`() {
+        val entity = HankeFactory.createEntity()
+
+        val result = HankeMapper.domainFrom(entity, mapOf(MOCK_ID to geometry))
+
+        assertThat(result).all {
+            prop(Hanke::id).isEqualTo(entity.id)
+            prop(Hanke::hankeTunnus).isEqualTo(entity.hankeTunnus)
+            prop(Hanke::onYKTHanke).isEqualTo(entity.onYKTHanke)
+            prop(Hanke::nimi).isEqualTo(entity.nimi)
+            prop(Hanke::kuvaus).isEqualTo(entity.kuvaus)
+            prop(Hanke::vaihe).isEqualTo(entity.vaihe)
+            prop(Hanke::suunnitteluVaihe).isEqualTo(entity.suunnitteluVaihe)
+            prop(Hanke::version).isEqualTo(entity.version)
+            prop(Hanke::createdAt).isEqualTo(entity.createdAt?.zonedDateTime())
+            prop(Hanke::createdBy).isEqualTo(entity.createdByUserId)
+            prop(Hanke::modifiedAt).isEqualTo(entity.modifiedAt?.zonedDateTime())
```

Comment:
There are no tests for `zonedDateTime()`, so these should use a literal value, not a conversion.

Reply:
Uses datefactory now.